### PR TITLE
Codearena-8: no pool validation allows rewards to be drained

### PR DESCRIPTION
**Pull request by grandizzy**
# Description of change
## High level
* See
https://github.com/code-423n4/2023-05-ajna-findings/issues/8
https://github.com/code-423n4/2023-05-ajna-findings/issues/64
https://github.com/code-423n4/2023-05-ajna-findings/issues/151
https://github.com/code-423n4/2023-05-ajna-findings/issues/207
  * exposed `PositionManager.isAjnaPool` external function to check if an address is a legit pool
  * update `RewardsManager.updateBucketExchangeRatesAndClaim` function to receive subset hash. check if address of pool is a legit Ajna adress by calling `Positionmanager.isAjnaPool`, otherwise revert with `NotAjnaPool` custom error
  * new `NotAjnaPool` custom error
  * unit tests update, POC mitigated

### CODE REVIEW

File: src/PositionManager.sol [hunk 1]
```diff
@@ -495,6 +495,14 @@ contract PositionManager is ERC721, PermitERC721, IPositionManager, Multicall, R
         );
     }
 
+    /// @inheritdoc IPositionManagerDerivedState
+    function isAjnaPool(
+        address pool_,
+        bytes32 subsetHash_
+    ) external override view returns (bool) {
+        return _isAjnaPool(pool_,subsetHash_);
```
Comment (EdNoepel) [inline src/PositionManager.sol]: nit: missing space between arguments

**Review by EdNoepel — APPROVED** [review]
Missing contract size and gas sections of PR template.

File: src/PositionManager.sol (hunk 1 above)
Comment (grandizzy) [reply src/PositionManager.sol]: fixed with https://github.com/ajna-finance/contracts/pull/824/commits/7b6a3b01fdc8beef56a495ed83ab084305c0ef0c

**Review by prat-gpt — APPROVED** [review]
LGTM! nice unit test.

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**